lib/cluster/etcd.go

```diff
@@ -805,3 +805,12 @@ func isSameDirSymlink(f fs.DirEntry, dir string) bool {
 	target, err := os.Readlink(filepath.Join(dir, f.Name()))
 	return err == nil && !strings.Contains(target, "/")
 }
+
+// GetStorageKey extracts the key from the source that contains the config prefix.
+func GetStorageKey(prefix, source string) (string, error) {
+	cfgPrefix := getConfigPrefix(prefix)
+	if !strings.HasPrefix(source, cfgPrefix) {
+		return "", fmt.Errorf("source must begin with: %s", cfgPrefix)
+	}
+	return source[len(cfgPrefix):], nil
```

Comment:
You could use `strings.CutPrefix` to do the thing.

Reply:
Thank you! Applied.